packages/core/saved-objects/core-saved-objects-server-internal/src/routes/index.ts

```diff
@@ -49,20 +50,30 @@ export function registerRoutes({
   migratorPromise: Promise<IKibanaMigrator>;
   kibanaVersion: string;
   kibanaIndex: string;
+  isServerless: boolean;
 }) {
   const router =
     http.createRouter<InternalSavedObjectsRequestHandlerContext>('/api/saved_objects/');
-  registerGetRoute(router, { config, coreUsageData, logger });
-  registerResolveRoute(router, { config, coreUsageData, logger });
-  registerCreateRoute(router, { config, coreUsageData, logger });
-  registerDeleteRoute(router, { config, coreUsageData, logger });
-  registerFindRoute(router, { config, coreUsageData, logger });
-  registerUpdateRoute(router, { config, coreUsageData, logger });
-  registerBulkGetRoute(router, { config, coreUsageData, logger });
-  registerBulkCreateRoute(router, { config, coreUsageData, logger });
-  registerBulkResolveRoute(router, { config, coreUsageData, logger });
-  registerBulkUpdateRoute(router, { config, coreUsageData, logger });
-  registerBulkDeleteRoute(router, { config, coreUsageData, logger });
+
+  const internalOnServerless = isServerless ? 'internal' : 'public';
+
+  registerGetRoute(router, { config, coreUsageData, logger, access: internalOnServerless });
+  registerResolveRoute(router, { config, coreUsageData, logger, access: internalOnServerless });
+  registerCreateRoute(router, {
+    config,
+    coreUsageData,
+    logger,
+    access: internalOnServerless,
+  });
+  registerDeleteRoute(router, { config, coreUsageData, logger, access: internalOnServerless });
+  registerFindRoute(router, { config, coreUsageData, logger, access: internalOnServerless });
+  registerUpdateRoute(router, { config, coreUsageData, logger, access: internalOnServerless });
+  registerBulkGetRoute(router, { config, coreUsageData, logger, access: internalOnServerless });
+  registerBulkCreateRoute(router, { config, coreUsageData, logger, access: internalOnServerless });
+  registerBulkResolveRoute(router, { config, coreUsageData, logger, access: internalOnServerless });
+  registerBulkUpdateRoute(router, { config, coreUsageData, logger, access: internalOnServerless });
+  registerBulkDeleteRoute(router, { config, coreUsageData, logger, access: internalOnServerless });
```

Comment:
I did not add tests for the existing public behaviour as I assume these tests already give us that coverage:

https://github.com/elastic/kibana/blob/e3ca2c59b0afc056d5f8ed77af1732961c2b190b/src/core/server/integration_tests/saved_objects/routes/bulk_get.test.ts#L85